Observer/TrackingSaveObserver.php

```diff
@@ -112,13 +112,13 @@ public function execute(Observer $observer)
             $order = $shipment->getOrder();
             $payment = $order->getPayment();
 
-            // If this is not a bolt payment, ignore it
-            if (!$payment || $payment->getMethod() != \Bolt\Boltpay\Model\Payment::METHOD_CODE) {
-                return;
+            $isNonBoltOrder = !$payment || $payment->getMethod() != \Bolt\Boltpay\Model\Payment::METHOD_CODE;
+            if ($isNonBoltOrder) {
+                $transactionReference = $payment->getAdditionalInformation('bolt_transaction_reference');
```

Comment:
Do you think it's possible that we don't have bolt_transaction_reference? For example, order was created before we updated plugin to the last version or bolt was temporary disabled.

Reply:
That’s a good point, I’ll handle this case